gh-95380: Removing the 1024 bytes limit in the fcntl_fcntl_impl and fcntl_ioctl_impl functions.

zenbooster opened the pull request:
Also, in the fcntl_ioctl_impl function, I removed adding zero to the end of the parameter for ioctl. Removed - works.

<!-- gh-issue-number: gh-95380 -->
* Issue: gh-95380
<!-- /gh-issue-number -->

zenbooster committed:
Поправил fcntl_fcntl_impl и fcntl_ioctl_impl. Для теста использовал два разных варианта исправления.

zenbooster committed:
Остановился на варианте с PyBytes_FromStringAndSize. В fcntl_ioctl_impl убрал buf[len] = '\0';

arhadthedev committed:
Убрать fcntl_fcntl_impl оставив fcntl_ioctl_impl

Пулл-реквест gh-95429 внёс по сути те же изменения, так что фокусируемся
на отсутствующем там fcntl_ioctl_impl.

gst commented on Modules/fcntlmodule.c:
```diff
@@ -228,14 +204,14 @@ fcntl_ioctl_impl(PyObject *module, int fd, unsigned int code,
-                PyErr_SetString(PyExc_ValueError,
-                                "ioctl string arg too long");
+
+            PyObject *o = PyBytes_FromStringAndSize(NULL, len);
```
it's not decref in the failure/error code path(s)..

also on line 248 : could not you directly returns it instead of re-creating another object ?

gst commented on Modules/fcntlmodule.c:
```diff
@@ -173,42 +169,22 @@ fcntl_ioctl_impl(PyObject *module, int fd, unsigned int code,
-            }
-            else {
-                ret = ioctl(fd, code, arg);
+            PyObject *o = PyBytes_FromStringAndSize(NULL, len);
```
should not be decref/freed in some code path(s) below ?

arhadthedev committed:
Перегенерировать обёртку, преобразующую питоновый вызов в сишный

Консольная команда, использованная для обновления:

    python Tools/clinic/clinic.py Modules/fcntlmodule.c

Если этого не сделать, автоматическая проверка пулл-реквеста увидит
рассинхронизацию, и у пул-реквеста в статусах будет крестик напротив
"Tests / Check if generated files are up to date".

arhadthedev committed:
Выполнил отмену изменений  fnctl_fnctl_impl до конца

arhadthedev committed:
Выполнил отмену изменений из более ранних коммитов

arhadthedev committed:
Добавить запись в журнал изменений

zenbooster committed:
fcntl_fcntl_impl: Returned as it was, at the request of GST:
https://github.com/python/cpython/issues/95380#issuecomment-1199496535
fcntl_ioctl_impl: add Py_DECREF.

zenbooster committed:
Merge pull request #1 from arhadthedev/fcntl_ioctl_impl

Remove `fcntl_fcntl_impl` keeping `fcntl_ioctl_impl`
